### ISSUE TITLE
feature request(browserstack): support BrowserStack SDK integration method

### ISSUE THREAD

**Issue by Kristinita** [seleniumbase/SeleniumBase#2401]
### 1. Summary

It would be nice if SeleniumBase will support [**BrowserStack SDK**](https://www.browserstack.com/docs/automate/selenium/browserstack-sdk/core-concepts) with a new BrowserStack configuration file [**`browserstack.yml`**](https://www.browserstack.com/docs/automate/selenium/sdk-config-generator).

### 2. Legacy method

Currently, SeleniumBase solely supports the old, “legacy” BrowserStack integration method.

![BrowserStack legacy capabilities](https://i.imgur.com/nTwB58C.png)

If I run my tests, use a command like this:

```text
pytest tests/test_kira_example.py --browser=remote --server=BROWSERSTACK_USERNAME:BROWSERSTACK_KEY@hub.browserstack.com --demo --reuse-session --port=80 --cap_file=kira_capabilities.py 
```

[**Tests run successfully on BrowserStack**](https://automate.browserstack.com/dashboard/v2/builds/805757c6af85c786ebfca977eb7a130015bbfa19/sessions/de4f32ead4caad32823c393d6f65cba2e6348698?auth_token=dba42bf4db1ecca8f7b64a979799ab4312c183430834a6917234fd7021ed782f).

(One problem — when all tests are completed, the remote BrowserStack session continues. I need to [**set timeout duration in BrowserStack settings**](https://www.browserstack.com/docs/automate/selenium/error-codes/browserstack-idle-timeout#increase-timeout-duration) or [**write code like `driver.quit()` especially for BrowserStack**](https://www.browserstack.com/docs/automate/selenium/error-codes/browserstack-idle-timeout#add-driverquit-command-in-test-script). I didn’t find in SeleniumBase an option like `--close-remote-session-after-all-tests`.)

### 3. BrowserStack SDK

The BrowserStack SDK is a new BrowserStack solution that [**allows to run tests without having to write code for BrowserStack**](https://www.browserstack.com/docs/automate/selenium/browserstack-sdk/benefits).

If we click on the [**link to BrowserStack capabilities**](https://www.browserstack.com/docs/automate/capabilities) in the SeleniumBase documentation, we see that BrowserStack offers by default the BrowserStack SDK, not Legacy integration method.

![BrowserStack SDK capabilities](https://i.imgur.com/ReiaZkx.png)

Currently, if I try to create a BrowserStack session with the `browserstack.yml` file, SeleniumBase shows an error:

```diff
- --cap_file=kira_capabilities.py
+ --cap_file=browserstack.yml
```

```python
C:\Python312\Lib\site-packages\seleniumbase\core\browser_launcher.py:1722: in get_remote_driver
    desired_caps = capabilities_parser.get_desired_capabilities(cap_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cap_file = 'browserstack.yml'

    def get_desired_capabilities(cap_file):
        if cap_file.endswith(".py"):
            capabilities = _parse_py_file(cap_file)
        elif cap_file.endswith(".json"):
            capabilities = _parse_json_file(cap_file)
        else:
>           raise Exception("\n\n`%s` is not a Python or JSON file!\n" % cap_file)
E           Exception:
E
E           `browserstack.yml` is not a Python or JSON file!
```

Thanks.

**Comment by mdmintz** [comment]
This was resolved in `4.22.4` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.22.4
See https://github.com/seleniumbase/SeleniumBase/issues/2403 for more details.
Happy New Year!

**Comment by mdmintz** [comment]
This was resolved in `4.22.5` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.22.5

To activate, include "browserstack" or "bstack" in your YAML file to set "bstack:options".
(Not case-sensitive.)

This is how it was done: https://github.com/seleniumbase/SeleniumBase/commit/50f8a0330eaec87893556e5c58c070c5ce53ea91
